src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotPolicyStep.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.common.exception.FeatureNotImplementedException;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.policy.PolicyService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteSnapshotPolicyStep extends DefaultUndoStep {
+  private static final Logger logger = LoggerFactory.getLogger(DeleteSnapshotPolicyStep.class);
+  private final DatasetService datasetService;
+  private final PolicyService policyService;
+  private final UUID snapshotId;
+
+  public DeleteSnapshotPolicyStep(
+      DatasetService datasetService, PolicyService policyService, UUID snapshotId) {
+    this.datasetService = datasetService;
+    this.policyService = policyService;
+    this.snapshotId = snapshotId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    FlightMap map = flightContext.getWorkingMap();
+    UUID datasetId = map.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
+    Dataset dataset = datasetService.retrieve(datasetId);
+    if (dataset.isSecureMonitoringEnabled()) {
+      try {
+        policyService.deletePao(snapshotId);
```

Comment:
What's the expected behavior if the snapshot in question has no associated policy?  One place this could happen is if we're deleting a snapshot that was created prior to our integration with TPS.

Reply:
If a policy isn't found when deleting a snapshot, we ignore the exception: https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/policy/PolicyService.java#L77

Reply:
Got it.  Since this behavior wasn't clear to me without reading the source code, what would you think about adding a docstring to the method and possibly renaming it to something like `deletePaoIfExists`?